### ISSUE TITLE
[Runtime Filter] The ```Unexpected empty values``` error when filter values is empty

### ISSUE THREAD

**Issue by elsa0520** [pingcap/tidb#45300]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create table
```
create table right_empty_table(k1 int);
alter table right_empty_table set tiflash replica 1;
create table left_table(k1 int, k2 int);
insert into left_table values (1,1);
 alter table left_table set tiflash replica 1;
set tidb_isolation_read_engines="tidb,tiflash";
```
2. enable runtime filter local
```
 set tidb_runtime_filter_mode=LOCAL;
``` 
3. query
```
mysql> select /*+ hash_join_build(right_empty_table) */ * from left_table , right_empty_table where left_table.k2=right_empty_table.k1;
ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1689068675331558971, local_query_id:11, server_id:2199946, start_ts:442779218824396801>,task_id:2>: Code: 0, e.displayText() = DB::Exception: Code: 0, e.displayText() = DB::Exception: Unexpected empty values, e.what() = DB::Exception,
```

### 2. What did you expect to see? (Required)

return empty result

### 3. What did you see instead (Required)

ERROR 1105 (HY000): other error for mpp stream: From MPP<query:<query_ts:1689068675331558971, local_query_id:11, server_id:2199946, start_ts:442779218824396801>,task_id:2>: Code: 0, e.displayText() = DB::Exception: Code: 0, e.displayText() = DB::Exception: Unexpected empty values, e.what() = DB::Exception,

### 4. What is your TiDB version? (Required)

v7.2 version and after

The TPC-DS query 37 and 61 have this  problem.

**Comment by elsa0520** [comment]
The runtime filter has not GA, so this bug almost no impact

**Comment by elsa0520** [comment]
In fact, when the data on the build side is empty and the Join type is inner join (that is, when the values of the runtime filter are empty), the probe side should be directly short-circuited in the hash join operator.
**This is the most correct and efficient way.**
But the short-circuit logic is designed to cross-node cancel, and the possible deadlock problem. So I will write an enhancement # here first, and we can consider the design later.

**Comment by elsa0520** [comment]
The root cause is that the build side of ```right_empty_table``` is empty. So that the filter values is empty.

When the filter  convert to RS operator, the storage engine forbidden the empty IN RS operator. So it throws the ```Unexpected empty values``` error.

The code of storage engine ```IN.h``` from ```tiflash/dbms/src/storage/delatmerge/Filter/In.h

```
    In(const Attr & attr_, const Fields & values_)
        : attr(attr_)
        , values(values_)
    {
        if (unlikely(values.empty()))
            throw Exception("Unexpected empty values");
    }
```

**Comment by elsa0520** [comment]
/remove-label may-affects-5.3 may-affects-5.4 may-affects-6.1 may-affects-6.5 may-affects-7.1

**Comment by elsa0520** [comment]
Closed pingcap/tiflash#7794
